### ISSUE TITLE
Stop parsing options after a subcommand, or add support for `--` to end options

### ISSUE THREAD

**Issue by ofisette** [codeandkey/mii#28]
On our HPC clusters, we call `mii select $1` from `command_not_found_handle`. However, if `$1` starts with `-`, this results in confusing results. For example, a user forgot to set a variable named `PROG` and got this message:

```
$PROG --version
[mii] loading mii build 1.0.4 Built on Fri 08 Oct 2021 03:10:49 PM EDT ...
Lmod has detected the following error:  The following module(s) are unknown:
"08" "Oct" "on" "EDT" "1.0.4" "Built" "03:10:49" "PM" "2021" "Fri" "build"

Please check the spelling or version number. Also try "module spider ..."
It is also possible your cache file is out-of-date; it may help to try:
  $ module --ignore_cache load "08" "Oct" "on" "EDT" "1.0.4" "Built" "03:10:49" "PM"
"2021" "Fri" "build"

Also make sure that all modulefiles written in TCL start with the string #%Module



-bash: --version: command not found
```

It would be helpful if `mii` stopped parsing options after a subcommand (like `select`) is given.

Alternatively, `mii` could implement `--` to explicitly end options (like `grep` does).

**Comment by codeandkey** [comment]
Thank you @ofisette for the report! I was able to reproduce this error and have implemented your solution. I think it's possible subcommands may have their own option sets in the future so I think using the '--' terminator is the best way to fix this. Fortunately as mii uses POSIX `getopt` this is already implemented, it needs only to be added to the handler. I added this in #29 and will merge shortly if this fixes the problem for you.
